lib/packwerk/layer/config.rb

```diff
@@ -0,0 +1,46 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Packwerk
+  module Layer
+    class Config
+      extend T::Sig
+
+      ARCHITECTURE_VIOLATION_TYPE = T.let('architecture', String)
+      ARCHITECTURE_ENFORCE = T.let('enforce_architecture', String)
+      LAYER_VIOLATION_TYPE = T.let('layer', String)
+      LAYER_ENFORCE = T.let('enforce_layers', String)
+
+      sig { void }
+      def initialize
+        @layers_key_configured = T.let(@layers_key_configured, T.nilable(T::Boolean))
+        @layers_list = T.let(@layers_list, T.nilable(T::Array[String]))
+      end
+
+      sig { returns(T::Array[String]) }
+      def layers_list
+        @layers_list ||= YAML.load_file('packwerk.yml')[layers_key] || []
+      end
+
+      sig { returns(T::Boolean) }
+      def layers_key_configured?
+        @layers_key_configured ||= YAML.load_file('packwerk.yml')['architecture_layers'].nil?
+      end
```

Comment:
An argument could be made for this class to be a singleton so it doesn't execute line `YAML.load_file` multiple times.

Reply:
I think that would be a bit cleaner